qep-342-charts.md

```diff
@@ -0,0 +1,123 @@
+# QGIS Enhancement: Charts
+
+**Date** 2025-06-20
+**Author** Mathieu Pellerin (@nirvn)
+**Contact** mathieu@opengis.ch
+**Version** QGIS 4.0
+
+## Summary
+
+This QEP aims at addressing a long-standing functionality gap within QGIS: 
+out-of-the-box charting capabilities. This crucial means to data visualization 
+has long been requested and would be a great addition to the layout designer. 
+
+### Motivations
+
+By creating QGIS-powered charting classes instead of relying on 3rd-party 
+libraries (such as D3), we would unlock powerful styling by leveraging our 
+pre-existing symbology capabilities. 
+
+Furthermore, the rendered graphs would benefit from a tighter integration 
+with map canvas and other layout items through sharing the same symbols, 
+drawn using the same text rendering engine, etc. This also means the resulting 
+charts retain their vector state when exporting content in supported formats 
+(PDFs, SVGs, etc.) and will be CMYK-compatible when running on Qt6 builds.
+
+Finally, developing a set of chart classes in src/core will be beneficial 
+to the QGIS server project as well as other sister projects such as QField 
+where solutions based on python plugins or QtWebkit-driven renderings simply 
+do not work.
+
+### Proposed solution
+
+The proposed solution will add charting classes aimed at behaving similarly
+to symbol classes with a render function taking a render context object.
+The charting capabilities will then be exposed through the layout designer
+via a new layout charts item.
+
+### Qgs{Line,Bar,Pie}Chart classes
+
+A new series of chart classes - QgsLineChart, QgsBarChart, and QgsPieChart -
```

Comment:
since the existing parent class is "QgsPlot", can we keep consistent naming and replace "chart" with "plot", or go with something like "QgsLineChartPlot" / "QgsPlotLineChart" ?

Reply:
Sure, that sounds good.